nilearn/plotting/html_surface.py

```diff
@@ -21,25 +21,45 @@ class SurfaceView(HTMLDocument):
 
 
 def _get_vertexcolor(surf_map, cmap, norm,
-                     absolute_threshold=None, bg_map=None):
+                     absolute_threshold=None, bg_map=None,
+                     bg_on_data=None, scale_bg_map=None, darkness=None):
     vertexcolor = cmap(norm(surf_map).data)
-    if absolute_threshold is None:
-        return to_color_strings(vertexcolor)
+
     if bg_map is None:
         bg_map = np.ones(len(surf_map)) * .5
         bg_vmin, bg_vmax = 0, 1
     else:
         bg_map = surface.load_surf_data(bg_map)
+
+    # scale background map if need be
+    if scale_bg_map:
         bg_vmin, bg_vmax = np.min(bg_map), np.max(bg_map)
-    bg_norm = mpl.colors.Normalize(vmin=bg_vmin, vmax=bg_vmax)
-    bg_color = plt.get_cmap('Greys')(bg_norm(bg_map))
-    vertexcolor[np.abs(surf_map) < absolute_threshold] = bg_color[
-        np.abs(surf_map) < absolute_threshold]
+        bg_norm = mpl.colors.Normalize(vmin=bg_vmin, vmax=bg_vmax)
+        bg_data = bg_norm(bg_map)
+    else:
+        bg_data = bg_map
+
+    if darkness is not None:
+        bg_data *= darkness
+    bg_color = plt.get_cmap('Greys')(bg_data)
+
+    # select vertices which are filtered out by the threshold
+    under_threshold = np.abs(surf_map) < (
+        absolute_threshold if absolute_threshold is not None else 0
+    )
+    # replace their color with the background color
+    vertexcolor[under_threshold] = bg_color[under_threshold]
+    # and merge background color with surface map color if need be
+    if bg_on_data:
+        vertexcolor[~under_threshold] = vertexcolor[~under_threshold] * \
+            bg_color[~under_threshold]
+
     return to_color_strings(vertexcolor)
 
 
 def one_mesh_info(surf_map, surf_mesh, threshold=None, cmap=cm.cold_hot,
                   black_bg=False, bg_map=None, symmetric_cmap=True,
+                  bg_on_data=False, scale_bg_map=True, darkness=1,
```

Comment:
Unrelated to this PR, but `one_mesh_info` and `full_brain_info` should be renamed with a leading "_", or added to the public API.

Reply:
I just added an underscore to both of them, we can still make them public later :)

Reply:
Update: we'll do this in another PR :)